sdk/hatchery/starter.go

```diff
@@ -226,28 +243,30 @@ func spawnWorkerForJob(ctx context.Context, h Interface, j workerStarterRequest)
 }
 
 // a worker name must be 60 char max, without '.' and '_', "/" -> replaced by '-'
-func generateWorkerName(hatcheryName string, isRegister bool, modelName string) string {
+const maxLength = 64
+
+func generateWorkerName(hatcheryName string, isRegister bool, modelName string) (string, error) {
 	prefix := ""
 	if isRegister {
 		prefix = "register-"
 	}
 
-	maxLength := 63
-	hName := hatcheryName + "-"
-	random := namesgenerator.GetRandomNameCDS(0)
-	workerName := fmt.Sprintf("%s%s-%s-%s", prefix, hatcheryName, modelName, random)
+	var nameFirstPart = fmt.Sprintf("%s%s", prefix, modelName)
+	var remainingLength = maxLength - len(nameFirstPart) - 1
 
-	if len(workerName) <= maxLength {
-		return slug.Convert(workerName)
+	if remainingLength <= 5 {
+		return "", errors.New("modelName too long")
 	}
-	if len(hName) > 10 {
-		hName = ""
-	}
-	workerName = fmt.Sprintf("%s%s%s-%s", prefix, hName, modelName, random)
+
+	random := namesgenerator.GetRandomNameCDSWithMaxLength(remainingLength)
+	workerName := fmt.Sprintf("%s-%s-%s", nameFirstPart, random, hatcheryName)
```

Comment:
Look like random is added here and also line268

Reply:
yes